### PR TITLE
DAOS-11224 datamover: accept user container label

### PR DESCRIPTION
**Pull request by daltonbohning**
Test-tag: pr datamover

- Fix incorrect handling of user-provided container label when creating
  a new container for fs copy and cont clone
- Add datamover test cases that use labels to verify this
- Refactor some datamover test cases

Required-githooks: true

Signed-off-by: Dalton Bohning <dalton.bohning@intel.com>

### CODE REVIEW

**Comment by daltonbohning** [comment]
@mchaarawi Most of this is ftest changes. If you're unfamiliar with that, it's really `daos_hdlr.c` I'd like your input on.

File: src/utils/daos_hdlr.c [hunk 1]
```diff
@@ -1949,27 +1949,41 @@ dm_connect(struct cmd_args_s *ap,
...
 				attr.da_props = props;
-				rc = dfs_cont_create(ca->dst_poh, &cuuid, &attr, NULL, NULL);
-				if (rc != 0) {
+				if (dst_cont_is_uuid)
+					rc = dfs_cont_create(ca->dst_poh, &cuuid, &attr,
+							     NULL, NULL);
+				else
+					rc = dfs_cont_create_with_label(ca->dst_poh, ca->dst_cont,
+									&attr, &cuuid, NULL, NULL);
+				if (rc) {
```
Comment (daltonbohning) [inline src/utils/daos_hdlr.c:1968]: This pattern is copied from mpifileutils, though I'm not sure if it's really how we should be doing this.

File: src/utils/daos_hdlr.c [hunk 2]
```diff
@@ -1949,27 +1949,41 @@ dm_connect(struct cmd_args_s *ap,
 			rc = -DER_NONEXIST;
 		}
 		if (rc == -DER_NONEXIST) {
-			uuid_t cuuid;
+			uuid_t	cuuid;
+			bool	dst_cont_is_uuid = true;
+
+			if (dst_cont_passed) {
+				rc = uuid_parse(ca->dst_cont, cuuid);
```
[the remark this answers is not in the document]
Comment (daltonbohning) [reply src/utils/daos_hdlr.c:1956]: I'm not opposed, though it is possible to create a container without a label. FWIW, I just realized that if a user passes a non-existent destination container UUID, the API ignores it and generates a new UUID. So at a bare minimum, this logic should return an error if the container UUID does not exist. I'll get feedback from the datamover group about only supporting labels.

[the remark this answers is not in the document]
Comment (daltonbohning) [reply src/utils/daos_hdlr.c:1956]: Or were you referring to removing UUID support only for non-existent containers?

[the remark this answers is not in the document]
Comment (mchaarawi) [reply src/utils/daos_hdlr.c:1956]: no i think we want everyone to use labels all the time and not use uuids anymore

**Review by shimizukko — DISMISSED** [review]

File: src/tests/ftest/util/data_mover_test_base.py [hunk 3]
```diff
@@ -388,10 +394,10 @@ def parse_create_cont_uuid(self, output):
             str: The parsed uuid.
 
         """
-        uuid_search = re.search(r"Successfully created container (.*-.*-.*-.*-.*)", output)
-        if not uuid_search:
+        label_search = re.search(r"Successfully created container (.*)", output)
+        if not label_search:
             self.fail("Failed to parse container uuid")
```
Comment (wiliamhuang) [inline src/tests/ftest/util/data_mover_test_base.py:399]: > self.fail("Failed to parse container uuid")

self.fail("Failed to parse container label")

Comment (daltonbohning) [reply src/tests/ftest/util/data_mover_test_base.py:399]: Ah, thanks!

**Review by wiliamhuang — DISMISSED** [review]

**Comment by daltonbohning** [comment]
Latest bug fix broke an NLT test:
https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-10023/12/tests

**Comment by daltonbohning** [comment]
One known test failure - https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-10023/16/tests
https://daosio.atlassian.net/browse/DAOS-10301

**Review by shimizukko — APPROVED** [review]

**Review by mchaarawi — APPROVED** [review]

**Pull request merged**